numpy/distutils/system_info.py

```diff
@@ -1683,6 +1689,7 @@ class blas_info(system_info):
     def calc_info(self):
         lib_dirs = self.get_lib_dirs()
         blas_libs = self.get_libs('blas_libs', self._lib_names)
+        blas_libs = self.get_libs('libraries', blas_libs)
```

Comment:
Does this merge the results or discard one half off them if both are specified?

Reply:
It has `libraries` precedence. So if only one is present, only that one will be used. I.e.:

```bash
[blas]
blas_libs = foo,bar

[blas]
libraries = foo,bar

[blas]
blas_libs = discarded
libraries = foo,bar
```
will all yield the same.

Reply:
In that case, you should emit an error if both are specified.

Reply:
If both are specified and we should raise an error it would require some additional restructuring which I don't think is necessary.
I would be against this since many of the `site.cfg` are undocumented features anyways. I.e. people would probably in some overlapping period expect the same `site.cfg` to work for both cases and when the older `*_libs` finally gets deprecated *then* one could do something.

Do you still think both should raise an error?

Reply:
Additionally I think this is of very minor relevance to 99.9% of users since the `site.cfg` is generally generated with one configuration in mind.

Reply:
>  are undocumented features anyways

All the more reason to raise a clear error if they do something that's a bad idea

> Do you still think both should raise an error?

Yes, silently discarding information should definitely raise an error.

Reply:
Ok, I'll fix this